packages/vite/src/node/plugins/manifest.ts

```diff
@@ -115,10 +130,25 @@ export function manifestPlugin(config: ResolvedConfig): Plugin {
         return manifestChunk
       }
 
+      const getFileName = (referenceId: string): string => {
+        try {
+          const fileName = this.getFileName(referenceId)
+          getReferenceIdToFileNameMapping(config).set(referenceId, fileName)
+          return fileName
+        } catch (error: unknown) {
+          const fallbackName =
+            getReferenceIdToFileNameMapping(config).get(referenceId)
+          if (fallbackName !== undefined) {
+            return fallbackName
+          }
+          throw error
+        }
+      }
+
       const fileNameToAssetMeta = new Map<string, GeneratedAssetMeta>()
       const assets = generatedAssets.get(config)!
       assets.forEach((asset, referenceId) => {
-        const fileName = this.getFileName(referenceId)
+        const fileName = getFileName(referenceId)
```

Comment:
I wonder if we can just ignore the error thrown from `this.getFileName`. IIUC the error happens if this `referenceId` is an id generated when generating with a different output option. The asset associated by that `referenceId` would be added to the manifest when this hook ran with that output option.

Reply:
Yes, the referenceId that caused the error was from a previous run. Just ignoring assets for which `this.getFileName` throws works well.